mne/viz/evoked.py

```diff
@@ -1844,11 +1879,78 @@ def _get_ci_function_for_evokeds(ci):
     return _ci_fun
 
 
+def _finish_styles_plot_comp_evoked(styles, colors, linestyles, conditions,
+                                    cmap):
+    """Finalize styles for plot_compare_evokeds."""
+    # Styles (especially color and linestyle) are pulled from a dict 'styles'.
+    # This dict has one entry per condition. Its color and linestyle entries
+    # are pulled from the 'colors' and 'linestyles' dicts via '/'-tag matching
+    # unless they are overwritten by entries from a user-provided 'styles'.
+
+    # first, copy to avoid overwriting
+    styles = deepcopy(styles)
+    colors = deepcopy(colors)
+    linestyles = deepcopy(linestyles)
+
+    # second, check if input is valid
+    if isinstance(styles, dict):
+        for style_ in styles:
+            if style_ not in conditions:
+                raise ValueError("Could not map between 'styles' and "
+                                 "conditions. Condition " + style_ +
+                                 " was not found in the supplied data.")
+
+    # third, color
+    if (colors is not None and not isinstance(colors, str) and
+            not isinstance(colors, dict) and len(colors) > 1):
+        colors = {condition: color for condition, color
+                  in zip(conditions, colors)}
+
+    cmap_label = ""
+    if cmap is not None:
+        if not isinstance(cmap, str) and len(cmap) == 2:
+            cmap_label, cmap = cmap
+
+    styles, the_colors, color_conds, color_order, colors_are_float =\
+        _setup_styles(conditions, styles, cmap, colors, linestyles)
+
+    return (styles, colors, the_colors, color_conds, color_order,
+            colors_are_float, cmap_label, cmap)
+
+
+def _plot_compare_evokeds(ax, data_dict, conditions, times, do_ci, ci_dict,
+                          styles, title, all_positive, topo):
+    """Plot evokeds (to compare them; with CIs) based on a data_dict."""
+    any_negative, any_positive = False, False
+    for condition in conditions:
+        # plot the actual data ('d') as a line
+        d = data_dict[condition].T
+        ax.plot(times, d, zorder=1000, label=condition, clip_on=False,
+                **styles[condition])
+        if np.any(d > 0) or all_positive:
+            any_positive = True
+        if np.any(d < 0):
```

Comment:
idem. You should not need to do this computation once any_negative has been set to True